packages/web3-react/src/components/AlephiumConnect.tsx

```diff
@@ -121,20 +117,76 @@ function tryGetNodeProvider(): NodeProvider | undefined {
   }
 }
 
-export const AlephiumConnectProvider: React.FC<{ children?: React.ReactNode }> = ({ children }) => {
+export const AlephiumConnectProvider: React.FC<{
+  network: NetworkId
+  addressGroup?: number
+  keyType?: KeyType
+  children?: React.ReactNode
+}> = ({ network, addressGroup, keyType, children }) => {
   // Only allow for mounting AlephiumConnectProvider once, so we avoid weird global
   // state collisions.
   const context = useContext(AlephiumConnectContext)
   if (context) {
     throw new Error('Multiple, nested usages of AlephiumConnectProvider detected. Please use only one.')
   }
 
-  const [account, setAccount] = useState<Account & { network: NetworkId }>()
+  const lastConnectedAccount = useMemo(() => {
+    const result = getLastConnectedAccount()
+    if (result === undefined) {
+      return undefined
+    }
+    if (
+      result.network === network &&
+      (addressGroup === undefined || result.account.group === addressGroup) &&
+      (keyType === undefined || result.account.keyType === keyType)
+    ) {
+      return result
+    }
+  }, [network, addressGroup, keyType])
+
+  const [account, setAccount] = useState<Account | undefined>(lastConnectedAccount?.account)
   const [signerProvider, setSignerProvider] = useState<SignerProvider | undefined>()
 
+  const updateAccount = useMemo(() => {
+    return (newAccount: Account | undefined) => {
+      setAccount((prev) => (prev?.address === newAccount?.address ? prev : newAccount))
+    }
+  }, [setAccount])
+
+  useEffect(() => {
+    const func = async () => {
+      if (lastConnectedAccount === undefined) return
+      const connector = getConnectorById(lastConnectedAccount.connectorId)
+      if (connector.autoConnect === undefined) return
+      const onDisconnected = () => {
+        updateAccount(undefined)
+        setSignerProvider(undefined)
+      }
+      const onConnected = (result: ConnectResult) => {
+        updateAccount(result.account)
+        setSignerProvider(result.signerProvider)
+      }
+
+      try {
+        const result = await connector.autoConnect({ network, addressGroup, keyType, onDisconnected, onConnected })
+        if (result === undefined) {
```

Comment:
When auto connect fails, we need to reset the state. This could happen when we reinstall the extension wallet.